### ISSUE TITLE
Docker images don't always pick up new packages

### ISSUE THREAD

**Issue by Marilyth** [Open-MSS/MSS#669]
As we could see with #665 our docker images don't seem to be installing the latest versions of packages reliably.
Issues that arise from updated packages thus don't always get caught.

It seems like the issue might come from using conda instead of mamba for installation, as these two don't behave the same way.
We might also have to run `conda/mamba update --all` to make sure new packages are installed.

**Comment by Marilyth** [comment]
For reference, this is `conda list` when installing through mamba
```
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
apipkg                    1.5              py39h06a4308_0  
appdirs                   1.4.4                      py_0  
argcomplete               1.12.2             pyhd3eb1b0_0  
argon2-cffi               20.1.0           py39he8ac12f_1  
basemap                   1.2.2            py39he68b33d_4    conda-forge
bcrypt                    3.2.0            py39he8ac12f_0  
blas                      1.1                    openblas    conda-forge
brotlipy                  0.7.0           py39h27cfd23_1003  
bzip2                     1.0.8                h7b6447c_0  
c-ares                    1.17.1               h27cfd23_0  
ca-certificates           2021.1.19            h06a4308_0  
certifi                   2020.12.5        py39h06a4308_0  
cffi                      1.14.4           py39h261ae71_0  
cftime                    1.4.1            py39hce5d2b2_0    conda-forge
chameleon                 3.6.2                      py_0  
chardet                   4.0.0           py39h06a4308_1003  
click                     7.1.2              pyhd3eb1b0_0  
conda                     4.9.2            py39h06a4308_0  
conda-package-handling    1.7.2            py39h27cfd23_1  
cryptography              3.3.1            py39h3c74f83_0  
curl                      7.71.1               he644dc0_8    conda-forge
cycler                    0.10.0           py39h06a4308_0  
dbus                      1.13.18              hb2f20db_0  
defusedxml                0.6.0                      py_0  
execnet                   1.8.0              pyhd3eb1b0_0  
expat                     2.2.10               he6710b0_2  
fastkml                   0.11                       py_1    conda-forge
flask                     1.1.2              pyhd3eb1b0_0  
flask-httpauth            4.2.0              pyhd8ed1ab_0    conda-forge
flask-socketio            4.3.2              pyhd3eb1b0_0  
flask-sqlalchemy          2.4.4                      py_0  
fontconfig                2.13.1               h6c09931_0  
freetype                  2.10.4               h5ab3b9f_0  
fs                        2.4.11           py39hde42818_2    conda-forge
fs.sshfs                  1.0.0              pyhd8ed1ab_1    conda-forge
fs.webdavfs               0.3.7            py39hde42818_1    conda-forge
fs_filepicker             0.3.7            py39hf3d152e_4    conda-forge
furl                      2.1.0                      py_0  
future                    0.18.2           py39h06a4308_1  
geos                      3.9.0                h9c3ff4c_0    conda-forge
gettext                   0.19.8.1             h9b4dc7a_1  
git                       2.30.0          pl5320h014a29a_0    conda-forge
gitdb                     4.0.5                      py_0  
gitpython                 3.1.12             pyhd3eb1b0_1  
glib                      2.66.6               h9c3ff4c_3    conda-forge
glib-tools                2.66.6               h9c3ff4c_3    conda-forge
gst-plugins-base          1.14.0               h8213a91_2  
gstreamer                 1.14.0               h28cd5cc_2  
hdf4                      4.2.13               h3ca952b_2  
hdf5                      1.10.6          nompi_h7c3c948_1111    conda-forge
humanfriendly             9.1              py39h06a4308_0  
icu                       58.2                 he6710b0_3  
idna                      2.10               pyhd3eb1b0_0  
importlib-metadata        3.4.0            py39hf3d152e_0    conda-forge
importlib_metadata        3.4.0                hd8ed1ab_0    conda-forge
importlib_resources       5.1.0            py39h06a4308_1  
isodate                   0.6.0                      py_1  
itsdangerous              1.1.0              pyhd3eb1b0_0  
jinja2                    2.11.3             pyhd3eb1b0_0  
jpeg                      9d                   h516909a_0    conda-forge
jplephem                  2.15               pyh2f6353c_0    conda-forge
kiwisolver                1.3.1            py39h2531618_0  
krb5                      1.17.2               h926e7f8_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.33.1               h53a641e_7  
libarchive                3.5.1                h899b81a_0    conda-forge
libblas                   3.9.0                8_openblas    conda-forge
libcblas                  3.9.0                8_openblas    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libedit                   3.1.20191231         h14c3975_1  
libev                     4.33                 h7b6447c_0  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
libgfortran-ng            7.5.0               h14aa051_18    conda-forge
libgfortran4              7.5.0               h14aa051_18    conda-forge
libglib                   2.66.6               h1f3bc88_3    conda-forge
libgomp                   9.3.0               h2828fa1_18    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
liblapack                 3.9.0                8_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h56d31a8_107    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.12          pthreads_hb3c22a3_1    conda-forge
libpng                    1.6.37               hbc83047_0  
libpq                     12.3                 hfd2b0eb_3    conda-forge
libsodium                 1.0.18               h7b6447c_0  
libsolv                   0.7.17               h780b84a_0    conda-forge
libssh2                   1.9.0                h1ba5d50_1  
libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
libtiff                   4.2.0                hdc55705_0    conda-forge
libuuid                   1.0.3                h1bed415_2  
libwebp-base              1.2.0                h27cfd23_0  
libxcb                    1.14                 h7b6447c_0  
libxml2                   2.9.10               hb55368b_3  
libxslt                   1.1.34               hc22bd24_0  
lxml                      4.6.2            py39h9120a33_0  
lz4-c                     1.9.2                heb0550a_3  
lzo                       2.10                 h7b6447c_2  
mamba                     0.7.12           py39h951de11_0    conda-forge
markdown                  3.3.3            py39h06a4308_0  
markupsafe                1.1.1            py39h27cfd23_0  
matplotlib                3.3.4            py39hf3d152e_0    conda-forge
matplotlib-base           3.3.4            py39h2fa2bec_0    conda-forge
multidict                 5.1.0            py39h27cfd23_2  
ncurses                   6.2                  he6710b0_1  
netcdf4                   1.5.5.1         nompi_py39h36800e2_101    conda-forge
numpy                     1.20.0           py39hdbf815f_0    conda-forge
olefile                   0.46                       py_0  
openblas                  0.3.12          pthreads_h43bd3aa_1    conda-forge
openssl                   1.1.1i               h27cfd23_0  
orderedmultidict          1.0.1                      py_0    conda-forge
owslib                    0.23.0             pyhd8ed1ab_0    conda-forge
packaging                 20.9               pyhd3eb1b0_0  
paramiko                  2.7.2                      py_0  
passlib                   1.7.4              pyh9f0ad1d_0    conda-forge
pcre                      8.44                 he6710b0_0  
perl                      5.32.0               h36c2ea0_0    conda-forge
pillow                    8.1.0            py39he98fc37_0  
pint                      0.16.1                     py_0    conda-forge
pip                       20.3.3           py39h06a4308_0  
proj                      7.2.0                h8b9fe22_0    conda-forge
property-cached           1.6.4                      py_0    conda-forge
psycopg2                  2.8.6            py39h3c74f83_1  
pycosat                   0.6.3            py39h27cfd23_0  
pycparser                 2.20                       py_2  
pygeoif                   0.7                        py_1    conda-forge
pymysql                   1.0.2            py39h06a4308_1  
pynacl                    1.4.0            py39he8ac12f_1  
pyopenssl                 20.0.1             pyhd3eb1b0_1  
pyparsing                 2.4.7              pyhd3eb1b0_0  
pyperclip                 1.8.1              pyhd3deb0d_0    conda-forge
pyproj                    3.0.0.post1      py39hb427721_0    conda-forge
pyqt                      5.9.2            py39h2531618_6  
pyshp                     2.1.3              pyhd3eb1b0_0  
pysocks                   1.7.1            py39h06a4308_0  
python                    3.9.1                hdb3f193_2  
python-dateutil           2.8.1              pyhd3eb1b0_0  
python-engineio           3.14.2             pyhd3eb1b0_0  
python-socketio           4.6.1              pyhd3eb1b0_0  
python_abi                3.9                      1_cp39    conda-forge
pytz                      2021.1             pyhd3eb1b0_0  
pyyaml                    5.4.1            py39h27cfd23_1  
qt                        5.9.7                h5867ecd_1  
readline                  8.1                  h27cfd23_0  
reproc                    14.2.1               h36c2ea0_0    conda-forge
reproc-cpp                14.2.1               h58526e2_0    conda-forge
requests                  2.25.1             pyhd3eb1b0_0  
ruamel_yaml               0.15.80         py39h3811e60_1004    conda-forge
scipy                     1.6.0            py39hf56f3a7_0  
setuptools                52.0.0           py39h06a4308_0  
sgp4                      2.15             py39h5472131_0    conda-forge
sip                       4.19.13          py39h2531618_0  
six                       1.15.0           py39h06a4308_0  
skyfield                  1.36               pyh44b312d_0    conda-forge
smmap                     4.0.0              pyhd3eb1b0_0  
sqlalchemy                1.3.23           py39h27cfd23_0  
sqlite                    3.33.0               h62c20be_0  
tk                        8.6.10               hbc83047_0  
tornado                   6.1              py39h27cfd23_0  
tqdm                      4.56.0             pyhd3eb1b0_0  
tzdata                    2020f                h52ac0ba_0  
unicodecsv                0.14.1           py39h06a4308_0  
urllib3                   1.26.3             pyhd3eb1b0_0  
validate_email            1.3                        py_3    conda-forge
webdavclient2             0.1.1                      py_0    conda-forge
werkzeug                  1.0.1              pyhd3eb1b0_0  
wheel                     0.36.2             pyhd3eb1b0_0  
xstatic                   1.0.2              pyh9f0ad1d_0    conda-forge
xstatic-bootstrap         4.5.3.1            pyhd3deb0d_0    conda-forge
xstatic-jquery            3.5.1.1          py39hf3d152e_0    conda-forge
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
zipp                      3.4.0              pyhd3eb1b0_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.5                h9ceee32_0
```

This is `conda list` when installing through conda
```
_libgcc_mutex             0.1                        main  
apipkg                    1.5              py39h06a4308_0  
appdirs                   1.4.4                      py_0  
argcomplete               1.12.2             pyhd3eb1b0_0  
argon2-cffi               20.1.0           py39he8ac12f_1  
basemap                   1.2.2            py39h523bb60_3    conda-forge
bcrypt                    3.2.0            py39he8ac12f_0  
blas                      2.11                   openblas    conda-forge
brotlipy                  0.7.0           py39h27cfd23_1003  
bzip2                     1.0.8                h7b6447c_0  
c-ares                    1.17.1               h27cfd23_0  
ca-certificates           2021.1.19            h06a4308_0  
certifi                   2020.12.5        py39h06a4308_0  
cffi                      1.14.4           py39h261ae71_0  
cftime                    1.3.1            py39h6323ea4_0  
chameleon                 3.6.2                      py_0  
chardet                   4.0.0           py39h06a4308_1003  
click                     7.1.2              pyhd3eb1b0_0  
cryptography              3.3.1            py39h3c74f83_0  
curl                      7.71.1               he644dc0_8    conda-forge
cycler                    0.10.0           py39h06a4308_0  
dbus                      1.13.18              hb2f20db_0  
defusedxml                0.6.0                      py_0  
execnet                   1.8.0              pyhd3eb1b0_0  
expat                     2.2.10               he6710b0_2  
fastkml                   0.11                       py_1    conda-forge
flask                     1.1.2              pyhd3eb1b0_0  
flask-httpauth            4.2.0              pyhd8ed1ab_0    conda-forge
flask-socketio            4.3.2              pyhd3eb1b0_0  
flask-sqlalchemy          2.4.4                      py_0  
fontconfig                2.13.1               h6c09931_0  
freetype                  2.10.4               h5ab3b9f_0  
fs                        2.4.11           py39hde42818_2    conda-forge
fs.sshfs                  1.0.0              pyhd8ed1ab_1    conda-forge
fs.webdavfs               0.3.7            py39hde42818_1    conda-forge
fs_filepicker             0.3.7            py39hf3d152e_4    conda-forge
furl                      2.1.0                      py_0  
future                    0.18.2           py39h06a4308_1  
geos                      3.8.1                he6710b0_0  
git                       2.23.0          pl526hacde149_0  
gitdb                     4.0.5                      py_0  
gitpython                 3.1.12             pyhd3eb1b0_1  
glib                      2.66.1               h92f7085_0  
gst-plugins-base          1.14.0               h8213a91_2  
gstreamer                 1.14.0               h28cd5cc_2  
hdf4                      4.2.13               h3ca952b_2  
hdf5                      1.10.6          nompi_h3c11f04_101    conda-forge
humanfriendly             9.1              py39h06a4308_0  
icu                       58.2                 he6710b0_3  
idna                      2.10               pyhd3eb1b0_0  
importlib-metadata        2.0.0                      py_1  
importlib_metadata        2.0.0                         1  
importlib_resources       5.1.0            py39h06a4308_1  
isodate                   0.6.0                      py_1  
itsdangerous              1.1.0              pyhd3eb1b0_0  
jinja2                    2.11.3             pyhd3eb1b0_0  
jpeg                      9d                   h36c2ea0_0    conda-forge
jplephem                  2.15               pyh2f6353c_0    conda-forge
kiwisolver                1.3.1            py39h2531618_0  
krb5                      1.17.1               h173b8e3_0  
lcms2                     2.11                 h396b838_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libblas                   3.8.0               11_openblas    conda-forge
libcblas                  3.8.0               11_openblas    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libedit                   3.1.20191231         h14c3975_1  
libev                     4.33                 h7b6447c_0  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
liblapack                 3.8.0               11_openblas    conda-forge
liblapacke                3.8.0               11_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h56d31a8_107    conda-forge
libnghttp2                1.41.0               hf8bcb03_2  
libopenblas               0.3.6                h5a2b251_2  
libpng                    1.6.37               hbc83047_0  
libpq                     12.2                 h20c2e04_0  
libsodium                 1.0.18               h7b6447c_0  
libssh2                   1.9.0                h1ba5d50_1  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_1  
libuuid                   1.0.3                h1bed415_2  
libxcb                    1.14                 h7b6447c_0  
libxml2                   2.9.10               hb55368b_3  
libxslt                   1.1.34               hc22bd24_0  
lxml                      4.6.2            py39h9120a33_0  
lz4-c                     1.9.3                h2531618_0  
markdown                  3.3.3            py39h06a4308_0  
markupsafe                1.1.1            py39h27cfd23_0  
matplotlib                3.3.2                h06a4308_0  
matplotlib-base           3.3.2            py39hd261b2b_0  
multidict                 5.1.0            py39h27cfd23_2  
ncurses                   6.2                  he6710b0_1  
netcdf4                   1.5.4           nompi_py39hb3be4b9_103    conda-forge
numpy                     1.19.4           py39h57d35e7_1    conda-forge
olefile                   0.46                       py_0  
openssl                   1.1.1i               h27cfd23_0  
orderedmultidict          1.0.1                      py_0    conda-forge
owslib                    0.22.0             pyhd3eb1b0_0  
packaging                 20.9               pyhd3eb1b0_0  
paramiko                  2.7.2                      py_0  
passlib                   1.7.4              pyh9f0ad1d_0    conda-forge
pcre                      8.44                 he6710b0_0  
perl                      5.26.2               h14c3975_0  
pillow                    8.1.0            py39he98fc37_0  
pint                      0.16.1                     py_0    conda-forge
pip                       20.3.3           py39h06a4308_0  
proj                      7.2.0                h8b9fe22_0    conda-forge
property-cached           1.6.4                      py_0    conda-forge
psycopg2                  2.8.6            py39h3c74f83_1  
pycparser                 2.20                       py_2  
pygeoif                   0.7                        py_1    conda-forge
pymysql                   1.0.2            py39h06a4308_1  
pynacl                    1.4.0            py39he8ac12f_1  
pyopenssl                 20.0.1             pyhd3eb1b0_1  
pyparsing                 2.4.7              pyhd3eb1b0_0  
pyperclip                 1.8.1              pyhd3deb0d_0    conda-forge
pyproj                    3.0.0.post1      py39hb427721_0    conda-forge
pyqt                      5.9.2            py39h2531618_6  
pyshp                     2.1.3              pyhd3eb1b0_0  
pysocks                   1.7.1            py39h06a4308_0  
python                    3.9.1                hdb3f193_2  
python-dateutil           2.8.1              pyhd3eb1b0_0  
python-engineio           3.14.2             pyhd3eb1b0_0  
python-socketio           4.6.1              pyhd3eb1b0_0  
python_abi                3.9                      1_cp39    conda-forge
pytz                      2021.1             pyhd3eb1b0_0  
pyyaml                    5.4.1            py39h27cfd23_1  
qt                        5.9.7                h5867ecd_1  
readline                  8.1                  h27cfd23_0  
requests                  2.25.1             pyhd3eb1b0_0  
scipy                     1.6.0            py39hf56f3a7_0  
setuptools                52.0.0           py39h06a4308_0  
sgp4                      2.10             py39ha90915f_3    conda-forge
sip                       4.19.13          py39h2531618_0  
six                       1.15.0           py39h06a4308_0  
skyfield                  1.36               pyh44b312d_0    conda-forge
smmap                     4.0.0              pyhd3eb1b0_0  
sqlalchemy                1.3.23           py39h27cfd23_0  
sqlite                    3.33.0               h62c20be_0  
tk                        8.6.10               hbc83047_0  
tornado                   6.1              py39h27cfd23_0  
tzdata                    2020f                h52ac0ba_0  
unicodecsv                0.14.1           py39h06a4308_0  
urllib3                   1.26.3             pyhd3eb1b0_0  
validate_email            1.3                        py_3    conda-forge
webdavclient2             0.1.1                      py_0    conda-forge
werkzeug                  1.0.1              pyhd3eb1b0_0  
wheel                     0.36.2             pyhd3eb1b0_0  
xstatic                   1.0.2              pyh9f0ad1d_0    conda-forge
xstatic-bootstrap         4.5.3.1            pyhd3deb0d_0    conda-forge
xstatic-jquery            3.5.1.1          py39hf3d152e_0    conda-forge
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
zipp                      3.4.0              pyhd3eb1b0_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.5                h9ceee32_0
```

**Comment by ReimarBauer** [comment]
please mention this situation also on the mamba issue tracker.

**Comment by ReimarBauer** [comment]
hmm, basemap build 4 is wrong. The rules should block access to it.

**Comment by ReimarBauer** [comment]
There may be a bias. We need a merge from stable to develop. in meta.yaml of stable is the fix geos <3.9.0

**Comment by Marilyth** [comment]
It also seems like we run into another issue related to versions. Which can be seen 
[here for develop](https://github.com/Open-MSS/dockerhub/runs/1901337903?check_suite_focus=true) and [here for stable](https://github.com/Marilyth/dockerhub/runs/1854030727?check_suite_focus=true)
The error originates form this line 
https://github.com/Open-MSS/MSS/blob/629887456abd9c0df9bc3ce402446ab86e186622/mslib/msui/_tests/test_mscolab.py#L54

We get a `Fatal Python error: Bus error` on test_mscolab.py when running the tests on python 3.8.6, but the issue vanishes when using python 3.9.1. This happens on both the stable and develop images and appeared only recently and seemingly without any new commits. I can not reproduce this on my local machine yet.
It is simple enough to update the python version on the docker images, but there might be something more serious going on.
